File: benchmarks/primes.js
// Prime Sieve Benchmark - Node.js

function isPrime(n) {
    if (n <= 1) return false;
    if (n === 2) return true;
    if (n % 2 === 0) return false;
    let divisor = 3;
    while (divisor * divisor <= n) {
        if (n % divisor === 0) return false;
        divisor += 2;
    }
    return true;
}

function countPrimes(limit) {
    let count = 0;
    for (let n = 2; n <= limit; n++) {
        if (isPrime(n)) count++;
    }
    return count;
}

console.log("=== Prime Sieve Benchmark (Node.js) ===");
const N = 10_000;

const start = performance.now();
const result = countPrimes(N);
const elapsed = (performance.now() - start) / 1000;

console.log(`Primes up to ${N}: ${result}`);
console.log(`Time: ${elapsed.toFixed(3)}s`);

File: benchmarks/primes.lua
-- Prime Sieve Benchmark - Lua

local function is_prime(n)
    if n <= 1 then return false end
    if n == 2 then return true end
    if n % 2 == 0 then return false end
    local divisor = 3
    while divisor * divisor <= n do
        if n % divisor == 0 then
            return false
        end
        divisor = divisor + 2
    end
    return true
end

local function count_primes(limit)
    local count = 0
    for n = 2, limit do
        if is_prime(n) then
            count = count + 1
        end
    end
    return count
end

print("=== Prime Sieve Benchmark (Lua) ===")
local N = 10000

local start = os.clock()
local result = count_primes(N)
local elapsed = os.clock() - start

print(string.format("Primes up to %d: %d", N, result))
print(string.format("Time: %.3fs", elapsed))
